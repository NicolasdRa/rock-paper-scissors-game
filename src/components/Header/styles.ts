import { makeStyles, createStyles, Theme } from '@material-ui/core/styles';

export const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    main: {
      display: 'flex',
      flexDirection: 'row',
      justifyContent: 'space-between',
      alignItems: 'center',
      backgroundColor: theme.palette.common.background,
      minHeight: 'max-content',
      width: '50%',
      // maxWidth: '30vw',
      border: '2px',
      borderStyle: 'solid',
      borderRadius: '1rem',
      borderColor: theme.palette.common.headerOutline,
      padding: '1rem 1.5rem',
      margin: '2rem auto',
    },

    titleBox: {
      display: 'flex',
      flexDirection: 'column',
      justifyContent: 'center',
      alignItems: 'flex-start',
      backgroundColor: theme.palette.common.background,
      minHeight: 'max-content',
      textTransform: 'uppercase',
      fontSize: '2rem',
    },

    title: {
      textTransform: 'uppercase',
      fontSize: '2rem',
      margin: '-4px',
      padding: 0,
      color: theme.palette.common.score,
      fontWeight: 600,
    },

    scoreBox: {
      display: 'flex',
      flexDirection: 'column',
      justifyContent: 'center',
      alignItems: 'center',
      backgroundColor: theme.palette.common.score,
      minHeight: 'max-content',
      textTransform: 'uppercase',
      padding: '1rem 2.5em',
      borderRadius: '6px',
    },
    scoreTitle: { fontSize: '1rem', color: theme.palette.common.scoreText },
    score: {
      fontSize: '3.5rem',
      fontWeight: 700,
      color: theme.palette.common.darkText,
    },
  }),
);
