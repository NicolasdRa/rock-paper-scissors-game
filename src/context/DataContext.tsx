import React, { createContext, useState } from 'react';

export type GlobalContext = {
  score: number;
  setScore: (score: number) => void;
  pick: string;
  setPick: (pick: string) => void;
};

export const DataContext = createContext<GlobalContext>({
  score: 0,
  setScore: () => {},
  pick: '',
  setPick: () => {},
});

interface Props {
  children?: React.ReactNode;
}

export const DataProvider: React.FC<Props> = ({ children }) => {
  const [score, setScore] = useState(12);
  const [pick, setPick] = useState('');

  return (
    <DataContext.Provider value={{ score, setScore, pick, setPick }}>
      {children}
    </DataContext.Provider>
  );
};
