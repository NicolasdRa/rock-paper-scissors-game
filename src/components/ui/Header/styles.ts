import { makeStyles, createStyles, Theme } from "@material-ui/core/styles";

export const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    header: {
      display: "flex",
      flexDirection: "row",
      justifyContent: "space-between",
      alignItems: "center",
      backgroundColor: theme.palette.common.background,
      width: "50vw",
      minHeight: "max-content",
      border: "2px",
      borderStyle: "solid",
      borderRadius: ".5rem",
      borderColor: theme.palette.common.headerOutline,
      padding: "1rem 1.5rem",
      margin: "1rem auto 2rem auto",

      [theme.breakpoints.down("md")]: {
        width: "90vw",
      },
    },

    titleBox: {
      display: "flex",
      flexDirection: "column",
      justifyContent: "center",
      alignItems: "flex-start",
      backgroundColor: theme.palette.common.background,
      minHeight: "max-content",
    },

    title: {
      textTransform: "uppercase",
      fontSize: "2rem",
      margin: "-4px",
      padding: 0,
      color: theme.palette.common.score,
      fontWeight: 600,

      [theme.breakpoints.down("md")]: {
        fontSize: "1.5rem",
      },
    },

    scoreBox: {
      display: "flex",
      flexDirection: "column",
      justifyContent: "center",
      alignItems: "center",
      backgroundColor: theme.palette.common.score,
      minHeight: "max-content",
      textTransform: "uppercase",
      padding: "1rem 2.5em",
      borderRadius: "6px",

      [theme.breakpoints.down("md")]: {
        padding: ".5rem 1.5rem",
        borderRadius: "6px",
      },
    },
    scoreTitle: {
      fontSize: "1rem",
      color: theme.palette.common.scoreText,

      [theme.breakpoints.down("md")]: {
        fontSize: ".8rem",
      },
    },
    score: {
      fontSize: "3.5rem",
      fontWeight: 700,
      color: theme.palette.common.darkText,

      [theme.breakpoints.down("md")]: {
        fontSize: "2.5rem",
      },
    },
  })
);
