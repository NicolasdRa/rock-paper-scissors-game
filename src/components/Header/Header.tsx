import React, { useContext } from "react";
import { DataContext } from "../../context/DataContext";

import { Box, Typography } from "@material-ui/core";

import { useStyles } from "./styles";
import theme from "../../theme";

export const Header: React.FC = () => {
  const classes = useStyles(theme);

  const { score } = useContext(DataContext);

  return (
    <Box className={classes.main}>
      <Box className={classes.titleBox}>
        <Typography variant="h3" className={classes.title}>
          Rock
        </Typography>
        <Typography variant="h3" className={classes.title}>
          Paper
        </Typography>
        <Typography variant="h3" className={classes.title}>
          Scissors
        </Typography>
      </Box>
      <Box className={classes.scoreBox}>
        <Typography variant="h3" className={classes.scoreTitle}>
          score
        </Typography>
        <Typography variant="h3" className={classes.score}>
          {score}
        </Typography>
      </Box>
    </Box>
  );
};
