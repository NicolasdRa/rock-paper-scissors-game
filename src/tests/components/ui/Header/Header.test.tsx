import "@testing-library/jest-dom";
import { mount } from "enzyme";
import { Header } from "../../../../components/ui/Header/Header";
import { DataContext } from "../../../../context/DataContext";

const contextValue = {
  setScore: jest.fn(),
  score: 5,
  setPick: jest.fn(),
  pick: "",
};

describe("tests on <Header/>", () => {
  const wrapper = mount(
    <DataContext.Provider value={contextValue}>
      <Header />
    </DataContext.Provider>
  );
  test("should render <Header/> correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should show score received from value set in Context", () => {
    expect(wrapper.find("h3").last().text()).toBe("5");
  });
});
