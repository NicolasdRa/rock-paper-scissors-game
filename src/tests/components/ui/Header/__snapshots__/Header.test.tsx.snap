// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on <Header/> should render <Header/> correctly 1`] = `
<div
  className="MuiBox-root MuiBox-root-7 makeStyles-header-1"
>
  <div
    className="MuiBox-root MuiBox-root-8 makeStyles-titleBox-2"
  >
    <h3
      className="MuiTypography-root makeStyles-title-3 MuiTypography-h3"
    >
      Rock
    </h3>
    <h3
      className="MuiTypography-root makeStyles-title-3 MuiTypography-h3"
    >
      Paper
    </h3>
    <h3
      className="MuiTypography-root makeStyles-title-3 MuiTypography-h3"
    >
      Scissors
    </h3>
  </div>
  <div
    className="MuiBox-root MuiBox-root-9 makeStyles-scoreBox-4"
  >
    <h3
      className="MuiTypography-root makeStyles-scoreTitle-5 MuiTypography-h3"
    >
      score
    </h3>
    <h3
      className="MuiTypography-root makeStyles-score-6 MuiTypography-h3"
    >
      5
    </h3>
  </div>
</div>
`;
